Client/WebFilters/CosmeticFiltersResourceDownloader.swift

```diff
@@ -41,23 +42,39 @@ class CosmeticFiltersResourceDownloader {
         if now.timeIntervalSince(lastFetchDate) >= fetchInterval {
             lastFetchDate = now
             
-            // MUST re-create the engine otherwise we get insane load times when calling:
-            // `engine_add_resources`
-            // This is because `engine_add_resources` will ADD resources, and not delete old ones
-            // Thus we get a huge amount of memory usage and slow down.
-            engine = AdblockRustEngine()
-            
-            loadDownloadedFiles()
-            downloadCosmeticSamples()
-            downloadResourceSamples()
+            Task {
```

Comment:
FYI this inherits from global executor which may or may not be `MainActor` if its used from within UIKit stuff, if you specifically want this task to run off `MainActor`, use `Task.detached`